File: sem4/mpp/project-java/services/src/main/java/services/IMasterService.java
package services;

import domain.*;
import domain.dto.ChildDTO;

import java.util.List;

public interface IMasterService {
	void register(String username, String password) throws Exception;
	void logIn(String username, String password, IUserObserver user) throws Exception;

	void logOut(IUserObserver user) throws Exception;
	List<ChildDTO> searchChildren(RaceDistance raceDistance, AgeGroup ageGroup) throws Exception;

	void signUpChild(String name, Integer age, RaceDistance raceDistance, AgeGroup ageGroup) throws Exception;
}

File: sem4/mpp/project-cs/services/IMasterService.cs
using project_cs.domain;

namespace project_cs.utils;

public interface IMasterService {
    public void Register(string username, string password);
    public void Login(string username, string password, IUserObserver user);
    public void Logout(IUserObserver user);
    public List<ChildDto> SearchChildren(RaceDistance raceDistance, AgeGroup ageGroup);
    public void SignUpChild(string name, int age, RaceDistance raceDistance, AgeGroup? ageGroup);
}